### PR TITLE
Add support for Xiaomi Mi Scale V1 and V2 

### PR DESCRIPTION
**Pull request by tiagofreire-pt**
## Description:

Regarding: https://github.com/esphome/esphome/pull/691

## Checklist:

  - [X] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.

### CODE REVIEW

**Comment by tiagofreire-pt** [comment]
The image could be this one: https://www.powerplanetonline.com/cdnassets/xiaomi-mi-scale-2-03_ad_l.jpg

**Review by OttoWinter — COMMENTED** [review]
1. The xiaomi_scale file has no extension, please rename it to `xiaomi_scale.rst`

2. The xiaomi_scale isn't added in the top-level index.rst file, please add it there.

3. This PR is created against the wrong branch, please create this PR against the `next` branch.

**Comment by tiagofreire-pt** [comment]
> 
> 
>     1. The xiaomi_scale file has no extension, please rename it to `xiaomi_scale.rst`
> 
>     2. The xiaomi_scale isn't added in the top-level index.rst file, please add it there.
> 
>     3. This PR is created against the wrong branch, please create this PR against the `next` branch.

I can't see that branch. Are you referring to the "master" one?

**Comment by tiagofreire-pt** [comment]
> 
> 
>     1. The xiaomi_scale file has no extension, please rename it to `xiaomi_scale.rst`
> 
>     2. The xiaomi_scale isn't added in the top-level index.rst file, please add it there.
> 
>     3. This PR is created against the wrong branch, please create this PR against the `next` branch.

I think I have done it. At least, as best as I could. :)

Thanks!

File: components/sensor/xiaomi_scale.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
+Xiaomi MiScale v1 and v2 BLE Sensor
+=========================
```
Comment (OttoWinter) [inline components/sensor/xiaomi_scale.rst]:
```suggestion
===================================
```

File: components/sensor/xiaomi_scale.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+Configuration variables:
+------------------------
+
+- **mac_address** (**Required**, MAC Address): The MAC address of the Xiaomi MiScale device.
+- **weight** (**Required**): The information for the weight sensor.
+
+  - **name** (**Required**, string): The name for the temperature sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
+
```
Comment (OttoWinter) [inline components/sensor/xiaomi_scale.rst:45]: This is missing two entries: `impedance` and `battery_level`

**Review by OttoWinter — COMMENTED** [review]
Please see the files tab, there are changes here that are not part of this changeset (for example changes in deep_sleep). Most likely due to wrong base branch, please fix that.

**Comment by glmnet** [comment]
Here are a few files not related to the xiaomi scale, also an entry in index is needed to link to this new page

File: components/sensor/xiaomi_scale.rst (hunk 2 above)
Comment (tiagofreire-pt) [reply components/sensor/xiaomi_scale.rst]: Done.

Comment (tiagofreire-pt) [reply components/sensor/xiaomi_scale.rst:45]: Done.

**Comment by tiagofreire-pt** [comment]
> Here are a few files not related to the xiaomi scale, also an entry in index is needed to link to this new page

Could you exemplify that, please?

Thanks.

**Comment by lupin-de-mid** [comment]
Probably: components/remote_transmitter.rst, components/deep_sleep.rst components/esp32_camera.rst guides/supporters.rst

**Pull request closed without merging**